### ISSUE TITLE
IPsec (ESP) performance drastically reduced from 4.10 to >=4.12

### ISSUE THREAD

**Issue by stuart-warren** [coreos/bugs#2294]
# Issue Report #

### Container Linux Version ###

```
$ cat /etc/os-release
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=1548.4.0
VERSION_ID=1548.4.0
BUILD_ID=2017-10-26-0356
PRETTY_NAME="Container Linux by CoreOS 1548.4.0 (Ladybug)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```

```
$ uname -a
Linux kubernetes-1511873754 4.13.9-coreos #1 SMP Thu Oct 26 03:35:38 UTC 2017 x86_64 Intel Core Processor (Haswell, no TSX) GenuineIntel GNU/Linux
```

### Environment ###

What hardware/cloud provider/hypervisor is being used to run Container Linux?

Openstack

kubernetes with weave networking

### Expected Behavior ###

network performance with ipsec offload between nodes should be good :)

### Actual Behavior ###

network performance with ipsec offload is terrible :(

On an encrypted (overlay) network:
bw = 262 KB/sec

On a non-encrypted (overlay) network:
bw = 113 MB/sec

### Reproduction Steps ###

  1. setup kubernetes with weave networking
  2. start 2 pods on different nodes
  3. use `ab` to test network throughput
  4. compare that to the network throughput on the same node
  5. ...
  6. cry at having lost 99% of your throughput

### Other Information ###

Believe we are hitting this kernel bug: https://bugzilla.kernel.org/show_bug.cgi?id=197513

There is a non committed kernel patch in one of the last comments.

Also see these bugs for further detail:
  * weaveworks/weave#3075
  * moby/moby#33133

**Comment by lucab** [comment]
Thanks for the report. I don't have a setup at hand to reproduce it. However according to the upstream b.k.o report this should be addressed by [this commit](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git/commit/net/xfrm/xfrm_output.c?h=v4.14.4&id=73b9fc49b4c0116a04eda3979f64ed9b540b153c), which is part of kernel v4.14.4.

According to your report you are currently running on 1548.4.0, which is a beta lagging behind by two release cycles and only shipping with kernel 4.13. Upgrading to a recent alpha or beta version should fix this issue.

**Comment by lucab** [comment]
This still affects stable so we are considering backporting that single commit. @stuart-warren were you able to confirm the issue is fixed on a newer kernel?

**Comment by stuart-warren** [comment]
I believe this has improved the situation, thank you.

**Comment by bgilbert** [comment]
This should be fixed in stable 1576.5.0, due shortly.  Thanks for the report.
